src/main/java/rx/internal/operators/OperatorConcat.java

```diff
@@ -172,7 +172,8 @@ void subscribeNext() {
                     child.onCompleted();
                 } else if (o != null) {
                     Observable<? extends T> obs = nl.getValue(o);
-                    currentSubscriber = new ConcatInnerSubscriber<T>(this, child, requested);
+                    currentSubscriber = new ConcatInnerSubscriber<T>(this, child);
+                    currentSubscriber.requestMore(requested);
```

Comment:
Sounds like you want to fix this issue using https://github.com/ReactiveX/RxJava/pull/2963 ?